pkg/api/versions.go

```diff
@@ -138,30 +145,50 @@ func yamlToStub(data []byte) ([]*Stub, error) {
 	return stubs, nil
 }
 
-// IsDeprecatedIn returns true if the version is deprecated in the targetVersion
+// IsDeprecatedIn returns true if the version is deprecated in the applicable targetVersion
 // Will return false if the targetVersion passed is not a valid semver string
-func (v *Version) isDeprecatedIn(targetVersion string) bool {
-	if !semver.IsValid(targetVersion) {
-		klog.V(3).Infof("targetVersion %s is not valid semVer", targetVersion)
-		return false
+func (v *Version) isDeprecatedIn(targetVersions map[string]string) bool {
+	for component, targetVersion := range targetVersions {
+		if !semver.IsValid(targetVersion) {
+			klog.V(3).Infof("targetVersion %s for %s is not valid semVer", targetVersion, component)
+			return false
+		}
 	}
+
 	if v.DeprecatedIn == "" {
 		return false
 	}
+
+	targetVersion, ok := targetVersions[v.Component]
+	if !ok {
+		klog.V(3).Infof("targetVersion missing for component %s", v.Component)
+		return false
+	}
+
 	comparison := semver.Compare(targetVersion, v.DeprecatedIn)
 	return comparison >= 0
 }
 
-// IsRemovedIn returns true if the version is deprecated in the targetVersion
+// IsRemovedIn returns true if the version is deprecated in the applicable targetVersion
 // Will return false if the targetVersion passed is not a valid semver string
-func (v *Version) isRemovedIn(targetVersion string) bool {
-	if !semver.IsValid(targetVersion) {
-		klog.V(3).Infof("targetVersion %s is not valid semVer", targetVersion)
-		return false
+func (v *Version) isRemovedIn(targetVersions map[string]string) bool {
+	for component, targetVersion := range targetVersions {
+		if !semver.IsValid(targetVersion) {
+			klog.V(3).Infof("targetVersion %s for %s is not valid semVer", targetVersion, component)
+			return false
+		}
 	}
+
 	if v.RemovedIn == "" {
 		return false
 	}
+
+	targetVersion, ok := targetVersions[v.Component]
```

Comment:
This check in both functions is not covered by tests. I think it would be good to get that in if possible